parsec/serde/fields.py

```diff
@@ -131,10 +130,10 @@ def _deserialize(self, value, attr, data):
     )
 
 
-def str_based_field_factory(value_type: Type) -> Type[Field]:
+def str_based_field_factory(value_type: Any) -> Type[Field]:
     assert isinstance(value_type, type)
 
-    def _serialize(self, value, attr, data):
+    def _serialize(self: Any, value: Any, attr: Any, data: Any) -> str | None:
```

Comment:
Same thing here from the docstring we can guess better types I think

```
Help on function _deserialize in module marshmallow.fields:

_deserialize(self, value, attr, data)
    Deserialize value. Concrete :class:`Field` classes should implement this method.

    :param value: The value to be deserialized.
    :param str attr: The attribute/key in `data` to be deserialized.
    :param dict data: The raw input data passed to the `Schema.load`.
    :raise ValidationError: In case of formatting or validation failure.
    :return: The deserialized value.
```